client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx

```diff
@@ -27,13 +35,13 @@ const plans: Step = function Plans( { navigation, flow } ) {
 	};
 
 	const getBackLabelText = () => {
-		if ( flow === DOMAIN_UPSELL_FLOW ) {
+		if ( flow === DOMAIN_UPSELL_FLOW || isHostingSiteCreationFlow( flow ) ) {
 			return __( 'Back' );
 		}
 	};
 
 	const shouldHideBackButton = () => {
-		if ( flow === DOMAIN_UPSELL_FLOW ) {
+		if ( flow === DOMAIN_UPSELL_FLOW || isHostingSiteCreationFlow( flow ) ) {
 			return false;
 		}
 		return true;
```

Comment:
Would it be clearer expressed like this?

```suggestion
		return flow !== DOMAIN_UPSELL_FLOW && ! isHostingSiteCreationFlow( flow );
```